### DISCUSSION TITLE
Leaderboards

### DISCUSSION THREAD

**Discussion by mithrand0** [ReactiveDrop/reactivedrop_public_src#179] [Polls]
Leaderboards and ranks are kinda the same thing from the point of functionality. Both let you compete against friends.

Currently, ranks are only available from 'approved' dedicated servers. In my opinion, it would be good to do the same with the Leaderboards. Also, leaderboards could use the same 'season' format as ranks, allowing new leaderboards every season.

**Comment by wawawawawawawa** [comment]
Imo it's not the same, leaderboard is something you should be able to do locally, for solo challenges and for lag.
There aren't much official servers and you don't want them filled up with solo challenges.
The overall increase in use of points server compared to other servers is high, because points. If you add normal leaderboard, it will make it even worse.

Also, for new players, not having their score updated if they create a lobby is a big wtf, and makes no sense.

Yes, leaderboard is cheated, but heck, points server too (client cheats), so that doesn't matter

Removing the option to go solo with bots (for leaderboard that only shows you and your friends anyway) is bad imo.

**Reply by wawawawawawawa** [reply]
Instead of preventing new players to "Create a Game" and get LB times (which again, LB showing only friends ingame), why not add times into the ranked stats?

If done that way, we could speedrun hunt every season (aka infinite retention, and u can tie some points/medals for speedrunning), and still have the cheat LB that would record the forever best time (and manually delete cheaters time like now)

**Comment by BenLubar** [comment]
We currently have 9023 leaderboards, and the hard limit is 10000. HoIAF uses a combination of steam stats and the stats website server to store data, but I don't think that would scale to the size of the full AS:RD playerbase. Plus, the server frequently goes down for reasons I'm still trying to figure out.

Making all leaderboards official-only would put a hard dependency on the stats website staying up, which is a single point of failure I'm not comfortable with adding.

**Reply by wawawawawawawa** [reply]
That means LB will get blanked out (on some challenges or whatnot) soon enough anyway since 1000 spots left?

If so, two possible ways to increase the number of spots : 
1) After X spot, you only save time in client (yes, it would mean ppl could locally change it, but it wouldn't change the rank, so wouldn't matter), and in website, you get only 1 entry for that Xth spot with loads of name.
2) Probably harder to do, but change the counting so that 1st team is 1 spot instead of 8 spots.

**Reply by BenLubar** [reply]
There are definitely leaderboards created by pirated games in there that can be safely discarded, and also leaderboards created by modded clients that are set to upload scores regardless of challenge or difficulty whitelisting.

I don't see us needing to delete any leaderboards that currently show up on the stats site, since a new map being created and uploaded to the workshop only adds a few leaderboard variants at most. If we end up getting a surge in popularity, we can look into other methods of storing leaderboard data, but the 9k leaderboards we currently have are probably going to stay at about that level for quite some time.

**Comment by mithrand0** [comment]
Steam discussion related for this https://steamcommunity.com/app/563560/discussions/2/3361397532252356972/

**Comment by mithrand0** [comment]
A decision has been made on this topic:

> Leaderboards are casual runs, designed to compare your time with your friends, they are not verified, and some people might have cheated them*.
>
> * If you get caught or reported, cheating might get you banned, see [https://reactivedrop.com/anticheat/]> (https://reactivedrop.com/anticheat) for details.
